src/index.js

```diff
@@ -1,207 +1,210 @@
+      (function(){function r(){if(!window.frames.__cmpLocator){if(document.body&&document.body.firstChild){var e=document.body;var t=document.createElement("iframe");t.style.display="none";t.name="__cmpLocator";t.title="cmpLocator";e.insertBefore(t,e.firstChild)}else{setTimeout(r,5)}}}function e(e,t,r){if(typeof r!=="function"){return}if(!window.__cmpBuffer){window.__cmpBuffer=[]}if(e==="ping"){r({gdprAppliesGlobally:window.gdprAppliesGlobally,cmpLoaded:false},true)}else{window.__cmpBuffer.push({command:e,parameter:t,callback:r})}}e.stub=true;function t(a){if(!window.__cmp||window.__cmp.stub!==true){return}if(!a.data){return}var n=typeof a.data==="string";var e;try{e=n?JSON.parse(a.data):a.data}catch(t){return}if(e.__cmpCall){var o=e.__cmpCall;window.__cmp(o.command,o.parameter,function(e,t){var r={__cmpReturn:{returnValue:e,success:t,callId:o.callId}};a.source.postMessage(n?JSON.stringify(r):r,"*")})}}if(typeof window.__cmp!=="function"){window.__cmp=e;if(window.addEventListener){window.addEventListener("message",t,false)}else{window.attachEvent("onmessage",t)}}r()})();(function(e){var t=e?e+"/":"";var r=document.createElement("script");r.id="spcloader";r.type="text/javascript";r.async=true;r.src=sdkPath+t+"loader.js?"+loaderParams;r.charset="utf-8";var a=document.getElementsByTagName("script")[0];a.parentNode.insertBefore(r,a)})(apiKey);
     }
   }
 
-  componentDidMount() {
-    this.init();
+  React.useEffect(() => {
+    init();
 
-    if(this.props.onReady) {
+    if(onReady) {
       window.didomiOnReady = window.didomiOnReady || [];
-      window.didomiOnReady.push(this.didomiOnReady.bind(this));
-    }
-  }
-
-  render() {
-    return null;
-  }
-};
+      window.didomiOnReady.push(handleDidomiOnReady);
+    }
+  }, []) 
+
+  return null;
+}
+
+DidomiSDK.propTypes = {
+  apiKey: PropTypes.string,
+  iabVersion: PropTypes.number,
+  noticeId: PropTypes.string,
+  config: PropTypes.object,
+  gdprAppliesGlobally: PropTypes.bool,
+  onReady: PropTypes.func,
+  onConsentChanged: PropTypes.func,
+  onNoticeShown: PropTypes.func,
+  onNoticeHidden: PropTypes.func,
+  onNoticeBackdropclick: PropTypes.func,
+  onNoticeClickAgree: PropTypes.func,
+  onNoticeClickMoreInfo: PropTypes.func,
+  onPreferencesClickAgreeToAll: PropTypes.func,
+  onPreferencesClickDisagreeToAll: PropTypes.func,
+  onPreferencesClickPurposeAgree: PropTypes.func,
+  onPreferencesClickPurposeDisagree: PropTypes.func,
+  onPreferencesClickViewVendors: PropTypes.func,
+  onPreferencesClickSaveChoices: PropTypes.func,
+  onPreferencesClickVendorAgree: PropTypes.func,
+  onPreferencesClickVendorDisagree: PropTypes.func,
+  onPreferencesClickVendorSaveChoices: PropTypes.func,
+  sdkPath: PropTypes.string
```

Comment:
We also need to add this property to the `index.d.ts` for typescript

Reply:
Done in https://github.com/didomi/react/pull/25/commits/93a33656f6d1080742f273a67e867eee73943d13